test/e2e/custom_route_test.go

```diff
@@ -0,0 +1,185 @@
+		Namespace: "openshift-authentication",
+		Name:      "oauth-openshift",
+		Hostname:  "foo.bar.com",
+	})
+	require.NoError(t, err)
+
+	// Check that the hostname was updated
+	err = checkRouteHostname(t, routeClient, "openshift-authentication", "oauth-openshift", "foo.bar.com")
+	require.NoError(t, err)
+
+	// Provide a custom secret that does not exist
+	err = getAndUpdateComponentRoute(t, configClient, &configv1.ComponentRouteSpec{
+		Namespace: "openshift-authentication",
+		Name:      "oauth-openshift",
+		Hostname:  "foo.bar.com",
+		ServingCertKeyPairSecret: configv1.SecretNameReference{
+			Name: "missing-secret",
+		},
+	})
+	require.NoError(t, err)
+
+	// Check that the route is set to defaults if a non-existant secret is provided
+	ingressConfig, err := configClient.ConfigV1().Ingresses().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	err = checkRouteHostname(t, routeClient, "openshift-authentication", "oauth-openshift", "oauth-openshift."+ingressConfig.Spec.Domain)
+	require.NoError(t, err)
+
+	// Present and valid secrets cause hostname to change
+	err = getAndUpdateComponentRoute(t, configClient, &configv1.ComponentRouteSpec{
+		Namespace: "openshift-authentication",
+		Name:      "oauth-openshift",
+		Hostname:  "foo.bar.com",
+		ServingCertKeyPairSecret: configv1.SecretNameReference{
+			Name: secret.Name,
+		},
+	})
+	require.NoError(t, err)
+
+	err = checkRouteHostname(t, routeClient, "openshift-authentication", "oauth-openshift", "foo.bar.com")
+	require.NoError(t, err)
+}
+
+func getAndUpdateComponentRoute(t *testing.T, configClient *configclient.Clientset, componentRoute *configv1.ComponentRouteSpec) error {
+	t.Helper()
+	return wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
+		ingressConfig, err := configClient.ConfigV1().Ingresses().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			t.Logf("Unable to retrieve ingress config: %v", err)
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
+
+		found := false
+		for i, _ := range ingressConfig.Spec.ComponentRoutes {
+			if ingressConfig.Spec.ComponentRoutes[i].Namespace == componentRoute.Namespace &&
+				ingressConfig.Spec.ComponentRoutes[i].Name == componentRoute.Name {
+				ingressConfig.Spec.ComponentRoutes[i] = *componentRoute
+				found = true
+			}
+		}
+		if !found {
+			ingressConfig.Spec.ComponentRoutes = append(ingressConfig.Spec.ComponentRoutes, *componentRoute)
+		}
+
+		ingressConfig, err = configClient.ConfigV1().Ingresses().Update(context.TODO(), ingressConfig, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	})
+}
+
+func checkRouteHostname(t *testing.T, routeClient *routeclient.Clientset, routeNamespace string, routeName string, hostname string) error {
+	t.Helper()
+	return wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
10 minutes is a bit too much IMO

Reply:
Reducing to 1 minute.

Reply:
I reduced all poll functions to a duration of 1 minute other than the `pollForCustomServingCertificates`, which can take a few minutes to take affect.